rs/nns/gtc/canister/canister.rs

```diff
@@ -179,7 +179,6 @@ fn forward_whitelisted_unclaimed_accounts() {
 }
 
 #[candid_method(update, rename = "forward_whitelisted_unclaimed_accounts")]
-#[allow(clippy::let_unit_value)] // clippy false positive
```

Comment:
Entraining this delete is fine even though it is tangential, because a) it is super small b) it is a substantive improvement, not just changing comment formatting c) it won't show up in blame, since it is an outright delete, not a modification.

Reply:
I added it in the previous rust version bump as there was a bug in clippy. We delete it now given that the false positive case here is no longer triggered in the upgraded clippy version.

The idea is to delete them in subsequent upgrades when the bug is fixed and the allow statement is no longer needed

See https://github.com/dfinity/ic/pull/642#discussion_r1694758879

Reply:
I was supporting this change. I pointed this out, because it is an exception to my general "these seem like tangential comment reformat changes" feedback..